lib/qbo_api.rb

```diff
@@ -112,26 +112,42 @@ def request(method, path:, entity: nil, payload: nil, params: nil)
   end
 
   def response(resp, entity: nil)
-    data = JSON.parse(resp.body)
-    if entity
-      entity_response(data, entity)
-    else
-      data
-    end
+    data = parse_response_body(resp)
+    entity ? entity_response(data, entity) : data
   rescue => e
-    # Catch fetch key errors and just return JSON
+    # NOTE(BF): Rescue all StandardError and return parsed JSON body
+    QboApi.logger.debug { "[Quickbooks] response parsing error: entity=#{entity.inspect} body=#{resp.body.inspect} exception=#{e.inspect}" }
     data
   end
 
+  def parse_response_body(resp)
+    body = resp.body
+    case resp.headers['Content-Type']
+    when /json/ then JSON.parse(body)
```

Comment:
Let's go with .to_json https://github.com/minimul/qbo_api/issues/58

Reply:
it's already to_json in this PR. This is the parsing :)

Reply:
With a little connection refactoring, we can add the parse json middleware and skip this entirely..

Reply:
That's right – it was `JSON.generate` that was replaced with `.to_json`. 

Disregard.